from PyQt4.QtGui import QMainWindow

from gen.ui_main import UiMainWindow


class MainView(QMainWindow):
    def __init__( self , main_controller ):
        QMainWindow.__init__(self)
        self.main_controller = main_controller
        self.ui = None
        self.widgets_actives = []
        self.widget_active = None
        self.build_ui()

    def build_ui(self):
        self.ui = UiMainWindow()
        self.ui.setup_ui(self)
        self.ui.treeWidget.itemClicked.connect(lambda: self.printer(self.ui.treeWidget.currentItem()))
        self.ui.actionHome.triggered.connect(self.action_home )
        self.ui.actionNew_project.triggered.connect(self.action_new_project)
        self.ui.actionOpen_project.triggered.connect(self.action_open_project)
        self.ui.actionSave.triggered.connect(self.action_save)
        self.ui.actionVulnerabilities.triggered.connect(self.action_vulnerabilities)
        self.ui.actionScan.triggered.connect(self.action_scan)
        self.ui.actionSettings.triggered.connect(self.action_settings)
        self.ui.actionAbout.triggered.connect(self.action_about)

    def printer(self, item):
        print item.text(0)

    def set_message_status(self, message):
        self.ui.statusbar.showMessage(message)

    def add_widget(self, widget):
        if not self.widget_active:
            self.widget_active = widget
            self.ui.verticalLayout.addWidget(self.widget_active)
            self.ui.frame.setLayout(self.ui.verticalLayout)
        elif widget is not self.widget_active:
            self.ui.verticalLayout.removeWidget(self.widget_active)
            self.widget_active.hide()
            self.widget_active = widget
            self.widget_active.show()
            self.ui.verticalLayout.addWidget(self.widget_active)
            self.ui.frame.setLayout(self.ui.verticalLayout)
        elif widget is self.widget_active:
            pass

    def action_home(self ):
        self.main_controller.action_home()

    def action_new_project(self):
        self.main_controller.action_new_project()

    def action_open_project(self):
        self.main_controller.action_open_project()

    def action_save(self):
        self.main_controller.action_save()

    def action_vulnerabilities(self):
        self.main_controller.action_vulnerabilities()

    def action_scan(self):
        self.main_controller.action_scan()

    def action_settings(self):
        self.main_controller.action_settings()

    def action_about(self):
        self.main_controller.action_about()

    # def closeEvent(self, event):
    #     result = QtGui.QMessageBox.question(self,
    #                                         "Confirm Exit",
    #                                         "Are you sure you want to exit ?",
    #                                         QtGui.QMessageBox.Yes | QtGui.QMessageBox.No)
    #     event.ignore()
    #     if result == QtGui.QMessageBox.Yes:
    #         event.accept()
